src/core/pal/pal.cpp

```diff
@@ -133,7 +133,7 @@ bool extractFeatCallback( FeaturePart *ft_ptr, void *ctx )
     Feats *ft = new Feats();
     ft->feature = ft_ptr;
     ft->shape = nullptr;
-    ft->lPos = lPos;
+    ft->candidates = lPos;
```

Comment:
Suggestion: also rename the const QList< LabelPosition * > lPos variable declared above to candidates.